### ISSUE TITLE
syntax propertization is not happening in right place

### ISSUE THREAD

**Issue by vspinu** [jrblevin/markdown-mode#387]
You cannot just rely on `syntax-propertize-extend-region-functions` to do the job. 

To see the problem go to `markdown-syntax-propertize` and insert `(message "(markdown-syntax-propertize %d %d)" start end)` somewhere. 

Then go to a markdow document inside a code block and delete a character. The syntax propertize is not triggered. This is really bad because this means that all markdown caches from the point onward are incorrect and will trigger out-of-range error in the font-lock at some point. 

One really needs to setup an `after-change-hook` to flush the caches.

**Comment by vspinu** [comment]
I think a simple fix would be to run it _always_ in the font-lock extension, now just on`(= end (point-max))` condition [here](https://github.com/jrblevin/markdown-mode/blob/master/markdown-mode.el#L1068).

**Comment by vspinu** [comment]
Actually something is wrong on my side with syntax flushing, or it might be a bug in emacs. Syntax-propertize is called from font-lock so there is no need to call it again on markdown side. I will dig further and will be back if it's indeed markdown issue. Sorry for the noise.

**Comment by vspinu** [comment]
I have tracked it down to what I believe to be emacs bug. Often syntax-ppss flush is not refreshed after a change. This leads to outdated markdown caches and results in weird font-lock errors (mostly end of buffer, or wrong type integer-or-marker nil errors). 

I started a [thread](https://lists.gnu.org/archive/html/emacs-devel/2019-03/msg00500.html) on emacs devel on this. Hopefully something could be done about it there.
